src/main/java/bio/terra/service/dataset/DatasetTable.java

```diff
@@ -76,4 +82,31 @@ public DatasetTable primaryKey(List<Column> primaryKey) {
         this.primaryKey = primaryKey;
         return this;
     }
+
+    public PartitionStrategy getPartitionStrategy() {
+        return partitionStrategy;
+    }
+
+    public DatasetTable partitionStrategy(PartitionStrategy strategy) {
```

Comment:
Another alternative would be to make a base class for partitionStrategy and build the right one.
You only need the details inside PDAO, I think.